### PR TITLE
Update to weave v0.21.0

### LINKED ISSUE #1182: Adapt to new weave VersionedIDRef format
The fields that currently use the `VersionedIDRef` will change to 16 bytes as described in https://github.com/iov-one/weave/issues/918.

For this we need to adapt 

```ts
function decodeVersionedId(versionedId: codecImpl.orm.IVersionedIDRef): VersionedId {
  return {
    id: decodeNumericId(ensure(versionedId.id, "id")),
    version: ensure(versionedId.version, "version"),
  };
}
```

and pull out the two components using

```ts
function decodeVersionedId(versionedId: Uint8Array): VersionedId {
  return {
    id: Uint64.fromBytesBigEndian(versionedId.slice(0, 8)).toNumber(),
    version: Uint64.fromBytesBigEndian(versionedId.slice(8, 16)).toNumber(),
  };
}
```

This requires the next breaking release of weave.

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #1182

Opening a draft PR while waiting for weave to be available from DockerHub.

### CODE REVIEW

File: packages/iov-bns/src/decode.ts [hunk 1]
```diff
@@ -107,6 +107,13 @@ function decodeVersionedId(versionedId: codecImpl.orm.IVersionedIDRef): Versione
   };
 }
 
+function decodeVersionedIdArray(versionedId: Uint8Array): VersionedId {
+  return {
```
Comment (webmaster128) [inline packages/iov-bns/src/decode.ts]: Can you throw an error if the input does not have the expected length?

File: packages/iov-bns/src/decode.ts [hunk 2]
```diff
@@ -107,6 +107,13 @@ function decodeVersionedId(versionedId: codecImpl.orm.IVersionedIDRef): Versione
...
+    id: Uint64.fromBytesBigEndian(versionedId.slice(0, 8)).toNumber(),
+    version: Uint64.fromBytesBigEndian(versionedId.slice(8, 16)).toNumber(),
```
Comment (webmaster128) [inline packages/iov-bns/src/decode.ts]: The final imlementation ended up encoding the versin as Uint32 (4 bytes): https://github.com/iov-one/weave/pull/950/files, so we can use `Uint32` instead

Comment (willclarktech) [reply packages/iov-bns/src/decode.ts]: To clarify: 12 bytes total (8 for id, 4 for version)?

Comment (webmaster128) [reply packages/iov-bns/src/decode.ts]: Exactly

**Comment by webmaster128** [comment]
In addition to the current diff, we need to configure the name/namespace patterns in the genesis file (see https://github.com/iov-one/weave/pull/980/files#diff-19980658f2035866d4b2bef354b2827eR35-R40)

**Pull request merged**